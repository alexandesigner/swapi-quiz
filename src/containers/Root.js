/**
 * Root
*/

import React from 'react'
import PropTypes from 'prop-types'
import { Provider } from 'react-redux'
import { BrowserRouter as Switch, Route } from 'react-router-dom'

import { GlobalStyle } from '../styles/globalStyles'

import App from './App'
import Quiz from './Quiz'
import Rank from './Rank'

const Root = ({ store }) => (
  <Provider store={store}>
    <Switch>
      <main>
        <GlobalStyle />
        <Route exact path="/" component={App} />
        <Route path="/quiz" component={Quiz} />
        <Route path="/rank" component={Rank} />
      </main>
    </Switch>
  </Provider>
)

Root.propTypes = {
  store: PropTypes.object.isRequired
}

export default Root