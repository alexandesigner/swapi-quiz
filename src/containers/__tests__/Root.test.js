import React from 'react';
import {shallow} from 'enzyme';
import Root from '../Root';

import configureStore from '../../redux/store';

describe('Root', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Root store={configureStore()} />)
    expect(wrapper).toMatchSnapshot()
  });
})