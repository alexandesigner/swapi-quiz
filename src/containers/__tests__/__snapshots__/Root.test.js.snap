// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Root
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <BrowserRouter>
        <Route
          component={[Function]}
          path="/"
        />
        <Route
          component={[Function]}
          path="/quiz"
        />
        <Route
          component={[Function]}
          path="/rank"
        />
      </BrowserRouter>,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            path="/"
          />,
          <Route
            component={[Function]}
            path="/quiz"
          />,
          <Route
            component={[Function]}
            path="/rank"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/quiz",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/rank",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <BrowserRouter>
          <Route
            component={[Function]}
            path="/"
          />
          <Route
            component={[Function]}
            path="/quiz"
          />
          <Route
            component={[Function]}
            path="/rank"
          />
        </BrowserRouter>,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/"
            />,
            <Route
              component={[Function]}
              path="/quiz"
            />,
            <Route
              component={[Function]}
              path="/rank"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/quiz",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/rank",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
